lib/components/viewers/related-stops-panel.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react'
+
+import {
+  getModeFromRoute,
+  getRouteIdForPattern,
+  getStopName,
+  getStopTimesByPattern,
+  routeIsValid
+} from '../../util/viewer'
+import Icon from '../narrative/mode-icon'
+import { ComponentContext } from '../../util/contexts'
+
+import NextArrivalForPattern from './next-arrival-for-pattern'
+import RelatedPanel from './related-panel'
+import ViewStopButton from './view-stop-button'
+
+class RelatedStopsPanel extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      expanded: false
+    }
+  }
+
+  static contextType = ComponentContext
+
+  _toggleExpandedView = () => {
+    this.setState({ expanded: !this.state.expanded })
+  }
+
+  _getStopIcon = (stop) => {
+    const { ModeIcon } = this.context
+    const modes = [...new Set(stop.routes?.map(getModeFromRoute))]
+    if (modes.length === 1) {
+      return <ModeIcon height={22} mode={modes[0]} width={22} />
+    }
+    return <Icon type='map-marker' />
+  }
+
+  render () {
+    const { expanded } = this.state
+    const {
+      homeTimezone,
+      nearbyStops,
+      stopViewerArriving,
+      stopViewerConfig,
+      timeFormat
+    } = this.props
+    return (
+      <RelatedPanel
+        count={nearbyStops?.length}
+        expanded={expanded}
+        onToggleExpanded={this._toggleExpandedView}
+        title='Related Stops'
+      >
+        <ul className='related-items-list list-unstyled'>
+          {nearbyStops?.map((stop, index) => {
+            const { id } = stop
+            if (!expanded && index >= 2) return null
+            const stopTimesByPattern = getStopTimesByPattern(stop)
+            const label = getStopName(stop)
+
+            return (
+              <li className='related-item' key={id}>
+                <div className='stop-label'>
```

Comment:
Would it be possible to highlight a stop on the map when hovering a related station? Otherwise it can be a very overwhelming list of stops. See https://github.com/opentripplanner/otp-react-redux/pull/441/files#diff-d608ab158dfe6f609b16e5d538ab034cf5c04aae377d466666bd51f4f86db84dR178 for a possible approach